src/Common/src/System/Drawing/KnownColorTable.cs

```diff
@@ -3,550 +3,293 @@
-            }
-        }
-
-        private static void InitColorTable()
-        {
-            int[] values = new int[(unchecked((int)KnownColor.MenuHighlight)) + 1];
-
-            // When Microsoft.Win32.SystemEvents is available, we can react to the UserPreferenceChanging events by updating
-            // SystemColors. In order to avoid a static dependency on SystemEvents since it is not available on all platforms 
-            // and as such we don't want to bring it into the shared framework. We use the desktop identity for SystemEvents
-            // since it is stable and will remain stable for compatibility whereas the core identity could change.
-            Type systemEventsType = Type.GetType("Microsoft.Win32.SystemEvents, Microsoft.Win32.SystemEvents", throwOnError: false);
-            EventInfo upEventInfo = systemEventsType?.GetEvent("UserPreferenceChanging", BindingFlags.Public | BindingFlags.Static);
-
-            if (upEventInfo != null)
-            {
-                // Delegate TargetType
-                Type userPrefChangingDelegateType = Type.GetType("Microsoft.Win32.UserPreferenceChangingEventHandler, Microsoft.Win32.SystemEvents", throwOnError: false);
-                Debug.Assert(userPrefChangingDelegateType != null);
+            Debug.Assert(color > 0 && color <= KnownColor.MenuHighlight);
 
-                if (userPrefChangingDelegateType != null)
-                {
-                    // we are using MethodInfo overload because it allows relaxed signature binding i.e. the types dont need to be exact match. It allows base classes as well.
-                    MethodInfo mi = typeof(KnownColorTable).GetMethod(nameof(OnUserPreferenceChanging), BindingFlags.NonPublic | BindingFlags.Static);
-                    Debug.Assert(mi != null);
+            return Color.IsKnownColorSystem(color)
+                ? GetSystemColorArgb(color)
+                : s_colorTable[(int)color - (int)KnownColor.Transparent];
+        }
 
-                    if (mi != null)
-                    {
-                        // Creating a delegate to use it as event handler.
-                        Delegate handler = Delegate.CreateDelegate(userPrefChangingDelegateType, mi);
-                        upEventInfo.AddEventHandler(null, handler);
-                    }
+#if FEATURE_WINDOWS_SYSTEM_COLORS
 
-                    // Retrieving getter of the category property of the UserPreferenceChangingEventArgs.
-                    Type argsType = Type.GetType("Microsoft.Win32.UserPreferenceChangingEventArgs, Microsoft.Win32.SystemEvents", throwOnError: false);
-                    s_categoryGetter = argsType?.GetMethod("get_Category", BindingFlags.Instance | BindingFlags.Public);
-                    Debug.Assert(s_categoryGetter != null);
-                }
-            }
+        private static readonly byte[] s_systemColorIdTable = new byte[]
+        {
+            // In order of definition in KnownColor enum
 
```

Comment:
Nit: extra line.

Reply:
I left a line here as the second line only applies to the first group of colors